ext/standard/basic_functions.c

```diff
@@ -4700,6 +4704,33 @@ PHP_FUNCTION(error_get_last)
 }
 /* }}} */
 
+/* {{{ proto bool error_clear_last()
+   Clear the last occurred error. Returns false if there hasn't been an error yet. */
+PHP_FUNCTION(error_clear_last)
+{
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "") == FAILURE) {
+		return;
+	}
+
+	if (PG(last_error_message)) {
+		PG(last_error_type) = 0;
+		PG(last_error_lineno) = 0;
+
+		free(PG(last_error_message));
+		PG(last_error_message) = NULL;
+
+		if (PG(last_error_file)) {
+			free(PG(last_error_file));
+			PG(last_error_file) = NULL;
+		}
+
+		RETURN_TRUE;
+	}
+
+	RETURN_FALSE;
```

Comment:
I think this function should be void return.  if there is error message, then clear it, if none.. it already in a clean state..